usr/lib/pkcs11/soft_stdll/soft_specific.c

```diff
@@ -2268,6 +2340,13 @@ CK_RV token_specific_sha_final(STDLL_TokData_t *tokdata, DIGEST_CONTEXT *ctx,
         return CKR_FUNCTION_FAILED;
     }
 
+    switch (ctx->mech.mechanism) {
+    case CKM_SHA512_224:
+    case CKM_SHA512_256:
+        memcpy(orig_out_data, temp_out_data, hlen);
```

Comment:
You may want to erase the non-output bytes from memory here: OPENSSL_cleanse(temp_out_data, sizeof(temp_out_data))

Reply:
fixed